From 495ff930588c5f66dceb5f02d3a5a9cbbc31706a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim-Philipp=20M=C3=BCller?= <tim@centricular.com>
Date: Thu, 13 Jun 2024 13:20:48 +0100
Subject: [PATCH] meson: fix build on iOS with host_system set to 'iOS'

As used in GStreamer's cerbero meta-build system for
historical reasons (should be moved to host_system=darwin
and subsystem=ios).
---
 meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 97bc9b1..7f6c712 100644
--- a/meson.build
+++ b/meson.build
@@ -279,7 +279,7 @@ if not asm_option.disabled()
                 if cpu_subfamily == 'x86'
                     nasm_args += '-DPREFIX'
                 endif
-            elif host_system == 'darwin'
+            elif host_system in ['darwin', 'ios']
                 nasm_args += ['-DPREFIX', '-DSTACK_ALIGNMENT=16']
             elif host_system in ['linux', 'sunos', 'haiku', 'gnu', 'android']
                 nasm_args += ['-DSTACK_ALIGNMENT=16']
@@ -296,7 +296,7 @@ if not asm_option.disabled()
     elif cpu_subfamily == 'aarch64'
         enable_asm = true
         conf.set('ARCH_AARCH64', 1)
-        if host_system == 'darwin'
+        if host_system in ['darwin', 'ios']
             asm_args += '-DPREFIX'
         endif
     else
-- 
2.45.2

